tests/zmq/test_gnmi_zmq.py

```diff
@@ -0,0 +1,130 @@
+import pytest
+import logging
+import random
+import time
+
+from tests.common.utilities import wait_until
+from tests.common.helpers.assertions import pytest_assert
+
+logger = logging.getLogger(__name__)
+
+
+pytestmark = [
+    pytest.mark.topology('any')
+]
+
+
+def save_reload_config(duthost):
+
+    def _check_process_ready(process_name):
+        pid = duthost.shell("pgrep {}".format(process_name), module_ignore_errors=True)["stdout"]
+        logger.debug("_check_orchagent_ready: {} PID {}".format(process_name, pid))
+        return pid != ""
+
+    result = duthost.shell("sudo config save -y", module_ignore_errors=True)
+    logger.debug("Save config: {}".format(result))
+    result = duthost.shell("sudo config reload -y -f", module_ignore_errors=True)
+    logger.debug("Reload config: {}".format(result))
+
+    # swss and gnmi container may take some time to stop after reload config command
+    time.sleep(5)
```

Comment:
>sleep [](http://example.com/codeflow?start=9&length=5)

The hardcode sleep seconds are risky and make the test flaky.

Reply:
Fixed